### DISCUSSION TITLE
Battery Temperature and charging current

### DISCUSSION THREAD

**Discussion by jHamshaw** [TonyM1958/HA-FoxESS-Modbus#25] [Q&A]
Hi Tony - since you made various updates your Fox ESS ModBus integration has been working great on my H1-5.0-E-G2 (Gen 2) inverter installed in February 2024.

However there seems to be a difference between what you include as BMS1 Temperature (currently showing at 19.9 degrees C via ModBus and the Fox ESS app  battery temperature currently showing at 8.5 degrees C.

As my charge rate appears to be affected by temperature as the weather gets colder, is it possible to "expose" the figure highlighted in the screenshot please?

My inverter and batteries share a small loft space with my gas boiler so I may well add a small electric space heater and automate based on battery temperature as I can't seem to get to 100% SoC during a 3-hour Agile window at present!....

There are some HA graphs and more discussion over here  too : https://foxesscommunity.com/viewtopic.php?p=5455#p5455

Thanks

Jim

![Screenshot 2024-11-21 at 12 48 13](https://github.com/user-attachments/assets/e6dd295c-7ca2-45c8-b885-070e05124935)

**Comment by TonyM1958** [comment]
The BMS temperature is the temperature of the electronics in the BMS and is up to 10C higher than the cell temperatures. The Fox app used to show the BMS temperature but now shows the lowest cell temperature.

The H1 inverter has a range of BMS voltage and temperature values available via modbus: BMS Cell mv High, BMS Cell mv Low, BMS Cell Temperature High, BMS Cell Temperature Low.

However, when we looked at H1-G2, these modbus values were not available and only the BMS temperature was available.

Newer firmware for KH inverters uses a new Modbus mapping and seems to have moved these registers to new addresses so it might be worth probing the holding registers at 37617, 37618, 37619 and 37620 to see if any values are returned. If they are, I could add them into the modbus mapping.

**Reply by jHamshaw** [reply]
OK thanks - so I can see these 👍 

modpoll 3.15 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright (c) 2002-2024 proconX Pty Ltd
Visit https://www.modbusdriver.com for Modbus libraries and tools.

Protocol configuration: MODBUS/TCP, FC3
Slave configuration...: address = 247, start reference = 37617 (PDU), count = 4
Communication.........: 192.168.1.150, port 502, t/o 1.00 s, poll rate 1000 ms
Data type.............: 16-bit register, holding register table

-- Polling slave...
[37617]: 147
[37618]: 101
[37619]: 34
[37620]: 33

**Reply by TonyM1958** [reply]
Thanks - the values are not quite what I would expect but hopefully we can make sense of them. The first 2 are the max and min cell temperature 14.7 and 10.1 degrees C in this case. Please can you check the app currently says the cell temperature is around 10.1?

The next 2 are the cell max and min volts. On H1, these values are in mV so have 4 decimal places e.g. 3341mv = 3.341v. In your case, it looks like these are only being rendered in volts to 1 decimal place so the values are 3.4v and 3.3v. That's OK as an indication but it's not really enough precision to do things like calculate cell imbalance.

**Reply by TonyM1958** [reply]
Also be useful to see if you get anything for these holding registers:

37624 may be battery SoH although often this is not available for ECS or just reads as 0.
37632 may be Residual Energy in kWh
37633 may be Design Capacity in Ah.
37635 may be Design Energy in kWh

**Reply by jHamshaw** [reply]
Apologies for the delay - I've been  digging out an 80w space heater and setting up in my small dormer loft/boiler/inverter/battery space.

Latest values are :

[37617]: 160
[37618]: 108
[37619]: 33
[37620]: 32

and Fox App V2 shows battery temperature at 10.8 degrees C

**Reply by TonyM1958** [reply]
Good, thats broadly consistent with the 10.1C seen earlier so that says these are most likely the same registers on H1-G2 as well as KH.

I will have a look at adding some register definitions to the Modbus file when I get chance and let you know.

**Comment by TonyM1958** [comment]
BTW, there is a short explanation here: https://github.com/TonyM1958/FoxESS-Cloud/wiki/Fox-App-v2#battery-info

**Comment by IanR007** [comment]
Hi Jim,

To clear one of your points the FoxCloud2.0 app reports for battery the lowest cell temperature in the battery stack, this is usually cell m1. Trust this helps your understanding?You may wish to acquaint yourself more about FoxEss products here; https://www.facebook.com/share/g/7EZ8AT4CChoBQFHy/

Regards,Ian Rawlinson

**Reply by jHamshaw** [reply]
Thanks Ian - all I really need to know is how and which battery temperature parameters in ModBus (once Tony can expose them) map to BMS Max Charge Current in ECS2900 V2 battery firmware 2.000. I can then set up a low-wattage space heater in my small dormer small loft that contains my gas boiler, inverter and battery stack and automate it in HA....
